### PR TITLE
Add CPU Monitor Class for Enhanced Debugging

### PR DESCRIPTION
**Pull request by Usamasource**
**Description:**

> This update introduces the CPUMonitor class, designed to periodically monitor CPU usage when Lithops is operating in debug mode (log_level=debug). The addition of this class aims to facilitate more in-depth performance analysis and troubleshooting.

**Key Features:**

- > Periodic Monitoring: The CPUMonitor class periodically checks CPU usage, aggregating data to provide insights into CPU performance during Lithops execution.
- > Debug Mode Exclusive: To minimize resource consumption, the CPU monitoring functionality is enabled only in debug mode. This ensures that the additional overhead of running a monitoring thread does not impact regular Lithops operations.

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by JosepSampe** [comment]
@Usamasource grate addition, thanks. But the `monitor.py` file is exclusively used to track object storage files (status & result), so the CPUMonitor class should be located in [lithops/worker/utils.py](https://github.com/lithops-cloud/lithops/blob/master/lithops/worker/utils.py), where we already have, for example, memory monitoring functions, among others.

**Comment by Usamasource** [comment]
Thank you for your guidance regarding the placement of the CPUMonitor class. Following your advice, I have moved the class to utils.py. I appreciate your feedback :)

File: lithops/worker/jobrunner.py [hunk 1]
```diff
@@ -20,16 +20,18 @@
...
 import inspect
 import requests
+import threading
 import traceback
 from pydoc import locate
 from distutils.util import strtobool
 
 from lithops.worker.utils import peak_memory
-
+from lithops.monitor import CPUMonitor
```
Comment (JosepSampe) [inline lithops/worker/jobrunner.py:34]: is this import still correct?

File: lithops/worker/jobrunner.py [hunk 2]
```diff
@@ -46,7 +48,10 @@
 
 logger = logging.getLogger(__name__)
 
-
+import threading
+import time
+import psutil
```
Comment (JosepSampe) [inline lithops/worker/jobrunner.py:53]: These 3 libraries are already imported above

**Review by JosepSampe — CHANGES_REQUESTED** [review]
I added some comments in the code. Besides that, is there any particular reason you didn't update the `requirements.txt` file of the GCP Functions, Azure Functions, and Aliyun functions?

File: lithops/worker/jobrunner.py (hunk 1 above)
Comment (Usamasource) [reply lithops/worker/jobrunner.py:34]: It is correct since it is used from the run function to monitor the execution. Do you agree?

Comment (Usamasource) [reply lithops/worker/jobrunner.py:34]: As for the psutil dependency in the requirements files, it was an oversight on my part. Sorry :).

Comment (JosepSampe) [reply lithops/worker/jobrunner.py:34]: As far as I can see you moved the CPUMonitor from the monitor.py to the worker/utils.py. So I would expect the import to be` from lithops.worker.utils import CPUMonitor`

Comment (Usamasource) [reply lithops/worker/jobrunner.py]: Yes, you're right! :) I have changed it. Sorry!

**Review by JosepSampe — COMMENTED** [review]
I was evaluating if this feature is put in the correct place, and I think it would make more sense to put the `CPUMonitor` in the `worker/handler.py` file, before and after starting the `jobrunner` process.

In this PR implementation, the `CPUMonitor` is in the `jobrunner.py` file, which is the main process that runs the user function. This means the `CPUMonitor` will disturb the user function every second to get the CPU consumption.

In contrast, the `handler.py` process is responsible for starting the `jobrunner` process, remaining idle until the `jobrunner` process finishes. This means if we put the `CPUMonitor` there, it won't disturb the main `jobrunner` process.

**Comment by Usamasource** [comment]
Your proposed change makes a lot of sense and is indeed the best solution to avoid disrupting the user's process. It aligns with my initial intention, but thanks to your recommendation, I now have a better understanding of how Lithops works :)

**Review by JosepSampe — APPROVED** [review]

**Review by JosepSampe — CHANGES_REQUESTED** [review]
Tests fail to run, please address the issues and verify that the changes work before submitting a commit.
Linting tests also fails. Fix python styling issues.

Now that you moved the CPUMonitor to the `handler.py`, you must change the `jobrunner.stats.write` to `call_status.add`

**Pull request closed without merging**